### PR TITLE
Fix queue element access

### PR DESCRIPTION
**Pull request by RRozak**
Currently, Verilator in some cases incorrectly classifies access to queue element as rvalue. It throws then compilation errors, because `at` and `atBack` methods are used instead of `atWrite` and `atWriteAppendBack`. It happens e.g. when a queue or dynamic array element is passed to an output port of a task or if it is a target of dynamic cast. I decided to compute the type of access using V3LinkLValue, as it is used for variable references and possibly covers also other cases.

### CODE REVIEW

File: src/V3AstNodeExpr.h [hunk 1]
```diff
@@ -4424,13 +4424,17 @@ class AstSelBit final : public AstNodePreSel {
     // Single bit range extraction, perhaps with non-constant selection or array selection
     // Gets replaced during link with AstArraySel or AstSel
     // @astgen alias op2 := bitp
+private:
+    VAccess m_access;  // Left hand side assignment
```
Comment (wsnyder) [inline src/V3AstNodeExpr.h:4428]: I'm ok with this for now, but noting mostly to myself that long term might be better to have a separate node type for LValue selects.

File: src/V3LinkLValue.cpp [hunk 2]
```diff
@@ -311,6 +311,7 @@ class LinkLValueVisitor final : public VNVisitor {
         }
     }
     void visit(AstNodePreSel* nodep) override {
+        if (AstSelBit* selbitp = VN_CAST(nodep, SelBit)) selbitp->access(m_setRefLvalue);
```
Comment (wsnyder) [inline src/V3LinkLValue.cpp:314]: But, if there's a select where the bit-to-select has itself a select, won't this break?  Please add a test e.g. `foo[sel_bit[3:2]] = bar`.

Nit:
```suggestion
        if (AstSelBit* const selbitp = VN_CAST(nodep, SelBit)) selbitp->access(m_setRefLvalue);
```

**Review by wsnyder — COMMENTED** [review]
Ooh, good debug & fix!

File: src/V3LinkLValue.cpp (hunk 2 above)
Comment (RRozak) [reply src/V3LinkLValue.cpp]: Done.

**Comment by RRozak** [comment]
@wsnyder, do you think it can be merged now?

**Pull request merged**